crates/runtime/src/catalogconnector/glue/provider.rs

```diff
@@ -0,0 +1,475 @@
+        self.inner.databases.keys().cloned().collect()
+    }
+
+    fn schema(&self, name: &str) -> Option<Arc<dyn datafusion::catalog::SchemaProvider>> {
+        if self.inner.databases.contains_key(name) {
+            let schema_provider = GlueSchemaProvider {
+                database: name.to_string(),
+                state: Arc::clone(&self.inner),
+            };
+            Some(Arc::new(schema_provider))
+        } else {
+            None
+        }
+    }
+}
+
+#[async_trait]
+impl RefreshableCatalogProvider for GlueCatalogProvider {
+    async fn refresh(&self) -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+        // TODO: #6012
+        Ok(())
+    }
+}
+
+impl GlueSchemaProvider {
+    async fn create_iceberg_provider(
+        &self,
+        name: &str,
+        table: &Table,
+    ) -> DFResult<Option<Arc<dyn TableProvider>>> {
+        let mut props = Vec::new();
+        if let Some(provider) = self.state.config.credentials_provider() {
+            let creds = provider
+                .provide_credentials()
+                .await
+                .map_err(|e| DataFusionError::External(Box::new(e)))?;
+            props.push((S3_ACCESS_KEY_ID, creds.access_key_id().to_string()));
+            props.push((S3_SECRET_ACCESS_KEY, creds.secret_access_key().to_string()));
+        }
+
+        let file_io = FileIOBuilder::new("s3")
+            .with_props(props)
+            .build()
+            .map_err(|e| {
+                DataFusionError::External(Box::new(super::Error::BuildFileIO { source: e }))
+            })?;
+
+        let metadata_location = get_metadata_location(table.parameters.as_ref(), name)
+            .map_err(|e| DataFusionError::External(Box::new(e)))?;
+
+        let input_file = file_io.new_input(&metadata_location).map_err(|e| {
+            DataFusionError::External(Box::new(super::Error::CreateFileInput {
+                source: e,
+                location: metadata_location.clone(),
+            }))
+        })?;
+
+        let metadata_content = input_file.read().await.map_err(|e| {
+            DataFusionError::External(Box::new(super::Error::ReadMetadata {
+                source: e,
+                location: metadata_location.clone(),
+            }))
+        })?;
+
+        let metadata = serde_json::from_slice::<TableMetadata>(&metadata_content).map_err(|e| {
+            DataFusionError::External(Box::new(super::Error::DeserializeMetadata { source: e }))
+        })?;
+
+        let identifier =
+            TableIdent::new(NamespaceIdent::new(self.database.clone()), name.to_string());
+
+        let table = IcebergTable::builder()
+            .file_io(file_io)
+            .metadata(metadata)
+            .identifier(identifier)
+            .build()
+            .map_err(|e| {
+                DataFusionError::External(Box::new(super::Error::BuildIcebergTable { source: e }))
+            })?;
+
+        let table_provider = IcebergTableProvider::try_new_from_table(table)
```

Comment:
This needs to use the variant of the IcebergTableProvider that takes in a catalog, otherwise it will not get updates as the table itself is updated.

Reply:
See https://github.com/spiceai/spiceai/issues/5697